### PR TITLE
compiler: bump up to nightly-2020-10-25 for new AllocRef trait and AllocError

### PR DESCRIPTION
**Pull request by dingelish**
follow up changes in rust's libcore:
- https://github.com/rust-lang/rust/commit/d9d02fa168016b5b5b2033a2964a723f447f94b0
- https://github.com/rust-lang/rust/commit/5829560a68d4bf5e29edfed0f9545d24e87ac3f5

### CODE REVIEW

File: sgx_alloc/src/system.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl System {
             old_size => {
                 let new_ptr = self.alloc_impl(new_layout, zeroed)?;
                 ptr::copy_nonoverlapping(ptr.as_ptr(), new_ptr.as_mut_ptr(), old_size);
-                self.dealloc(ptr, old_layout);
+                core::alloc::AllocRef::dealloc(self, ptr, old_layout);
```
Comment (dingelish) [inline sgx_alloc/src/system.rs]: @volcano0dr need your confirmation here

File: sgx_alloc/src/system.rs [hunk 2]
```diff
@@ -109,55 +109,55 @@ impl System {
...
         ptr: NonNull<u8>,
         old_layout: Layout,
         new_layout: Layout,
-    ) -> Result<NonNull<[u8]>, AllocErr> {
+    ) -> Result<NonNull<[u8]>, AllocError> {
          match new_layout.size() {
              // SAFETY: conditions must be upheld by the caller
              0 => {
-                 self.dealloc(ptr, old_layout);
+                 core::alloc::AllocRef::dealloc(self, ptr, old_layout);
```
Comment (dingelish) [inline sgx_alloc/src/system.rs]: @volcano0dr need your confirmation here

File: sgx_alloc/src/system.rs [hunk 3]
```diff
@@ -176,9 +176,9 @@ unsafe impl AllocRef for System {
              // `new_ptr`. Thus, the call to `copy_nonoverlapping` is safe. The safety contract
              // for `dealloc` must be upheld by the caller.
              new_size => {
-                 let new_ptr = self.alloc(new_layout)?;
+                 let new_ptr = core::alloc::AllocRef::alloc(self, new_layout)?;
```
Comment (dingelish) [inline sgx_alloc/src/system.rs]: @volcano0dr need your confirmation here

File: sgx_alloc/src/system.rs [hunk 4]
```diff
@@ -176,9 +176,9 @@ unsafe impl AllocRef for System {
...
                  ptr::copy_nonoverlapping(ptr.as_ptr(), new_ptr.as_mut_ptr(), new_size);
-                 self.dealloc(ptr, old_layout);
+                 core::alloc::AllocRef::dealloc(self, ptr, old_layout);
```
Comment (dingelish) [inline sgx_alloc/src/system.rs]: @volcano0dr need your confirmation here

File: sgx_alloc/src/system.rs [hunk 5]
```diff
@@ -233,7 +233,7 @@ mod platform {
             if layout.align() <= MIN_ALIGN && layout.align() <= layout.size() {
                 libc::calloc(layout.size(), 1) as *mut u8
             } else {
-                let ptr = self.alloc(layout);
+                let ptr = GlobalAlloc::alloc(self, layout);
```
Comment (dingelish) [inline sgx_alloc/src/system.rs:236]: @volcano0dr need your confirmation here

**Comment by dingelish** [comment]
@jasl please take a look. thanks!

**Review by dingelish — COMMENTED** [review]
@volcano0dr please help me check if my changes are correct

**Comment by jasl** [comment]
> @jasl please take a look. thanks!

GRRRRRRRRREEEEEEEEEEAAAAAAAAAT 👍

**Comment by jasl** [comment]
I tried this PR, LGTM

**Pull request merged**